core/src/main/java/org/elasticsearch/transport/local/LocalTransport.java

```diff
@@ -288,6 +294,9 @@ private void handleRequest(StreamInput stream, long requestId, LocalTransport so
             if (reg == null) {
                 throw new ActionNotFoundTransportException("Action [" + action + "] not found");
             }
+            // reserve here in order be able to send a response in case we break the limit
+            inFlightRequestsBreaker.addEstimateBytesAndMaybeBreak(messageLengthBytes, "Node in-flight requests");
```

Comment:
can the label be `<transport_request>` ? seems to be more consistent with other stuff.